utils/self-extracting-executable/decompressor.cpp

```diff
@@ -382,6 +405,52 @@ int main(int/* argc*/, char* argv[])
     else
         name = file_path;
 
+    uint32_t inode = get_inode(self);
```

Comment:
this branch involves execv which will make this function no-return, besides it will require many parameters including argv - I think it's more clear and straight forward (as much as it can be) this way...